packages/client-sdk-nodejs/src/internal/storage-data-client.ts

```diff
@@ -0,0 +1,337 @@
+      clientFactoryFn: () =>
+        new store.StoreClient(
+          this.credentialProvider.getStorageEndpoint(),
+          this.credentialProvider.isStorageEndpointSecure()
+            ? ChannelCredentials.createSsl()
+            : ChannelCredentials.createInsecure(),
+          channelOptions
+        ),
+      loggerFactory: this.configuration.getLoggerFactory(),
+      maxIdleMillis: this.configuration
+        .getTransportStrategy()
+        .getMaxIdleMillis(),
+    });
+
+    const context: MiddlewareRequestHandlerContext = {};
+    context[CONNECTION_ID_KEY] = dataClientID;
+    this.interceptors = this.initializeInterceptors(
+      this.configuration.getLoggerFactory(),
+      this.configuration.getMiddlewares(),
+      context
+    );
+  }
+
+  close() {
+    this.logger.debug('Closing storage data clients');
+    this.clientWrapper.getClient().close();
+  }
+
+  private validateRequestTimeout(timeout?: number) {
+    this.logger.debug(`Request timeout ms: ${String(timeout)}`);
+    if (timeout !== undefined && timeout <= 0) {
+      throw new InvalidArgumentError(
+        'request timeout must be greater than zero.'
+      );
+    }
+  }
+
+  private initializeInterceptors(
+    _loggerFactory: MomentoLoggerFactory,
+    middlewares: Middleware[],
+    middlewareRequestContext: MiddlewareRequestHandlerContext
+  ): Interceptor[] {
+    const headers = [
+      new Header('Authorization', this.credentialProvider.getAuthToken()),
+      new Header('Agent', `nodejs:store:${version}`),
+    ];
+    return [
+      middlewaresInterceptor(
+        _loggerFactory,
+        middlewares,
+        middlewareRequestContext
+      ),
+      new HeaderInterceptorProvider(headers).createHeadersInterceptor(),
+      ClientTimeoutInterceptor(this.requestTimeoutMs),
+    ];
+  }
+
+  private createMetadata(storeName: string): Metadata {
+    const metadata = new Metadata();
+    metadata.set('store', storeName);
+    return metadata;
+  }
+
+  public async get(
+    storeName: string,
+    key: string
+  ): Promise<StorageGet.Response> {
+    try {
+      validateStoreName(storeName);
+    } catch (err) {
+      return this.cacheServiceErrorMapper.returnOrThrowError(
+        err as Error,
+        err => new StorageGet.Error(err)
+      );
+    }
+    this.logger.trace(
+      `Issuing 'get' request; store: ${storeName}, key: ${key}`
+    );
+    return await this.sendGet(storeName, key);
+  }
+
+  private async sendGet(
+    storeName: string,
+    key: string
+  ): Promise<StorageGet.Response> {
+    const request = new store._StoreGetRequest({
+      key: key,
+    });
+    const metadata = this.createMetadata(storeName);
+    return await new Promise((resolve, reject) => {
+      this.clientWrapper.getClient().Get(
+        request,
+        metadata,
+        {
+          interceptors: this.interceptors,
+        },
+        (err: ServiceError | null, resp) => {
+          const value = resp?.value?.value;
```

Comment:
looks like the JS protos also return Get value as an optional value?

Reply:
We currently resolve the none value to `UnknownError`:
```
case 'none': {
                return resolve(
                  new StorageGet.Error(
                    new UnknownError(
                      'StorageGet responded with an unknown result'
                    )
                  )
                );
              }
```

@malandis Based on the discussion today, do we resolve it to just returning the optional value like `resp?.value?`?